pkg/controller.v2/controller_tfjob.go

```diff
@@ -114,3 +115,38 @@ func (tc *TFJobController) deletePodsAndServices(tfJob *tfv1alpha2.TFJob, pods [
 	}
 	return nil
 }
+
+func (tc *TFJobController) cleanupTFJob(tfJob *tfv1alpha2.TFJob) error {
+	currentTime := time.Now()
+	if tfJob.Spec.TTLAfterFinished == nil {
+		// do nothing if the cleanup delay is not set
+		return nil
+	}
+	duration, err := time.ParseDuration(string(*tfJob.Spec.TTLAfterFinished))
+	if err != nil {
+		// do nothing if the duration string is not parsable
+		log.Warnf("Parse CleanDelay duration error: %v.", err)
+		return nil
+	}
+	if currentTime.After(tfJob.Status.CompletionTime.Add(duration)) {
+		err := tc.deleteTFJobHandler(tfJob)
+		if err != nil {
+			log.Warnf("Cleanup TFJob error: %v.", err)
+			return err
+		}
+		return nil
+	}
+	go func() {
```

Comment:
Instead of launching a background thread, could we just enqueue a work item into the workequeue 
https://github.com/kubeflow/tf-operator/blob/master/pkg/controller.v2/controller.go#L340
so that we will call reconcile after ~duration seconds.

The workqueue will automatically take care of combining events and rate limiting things.

Arguably we could just rely on the fact that reconcile gets called periodically.Which would mean that we could wait up to ReconcilePeriod extra time before deleting the job.

I think that would be fine but we should make a note of that in the field documentation.